### PR TITLE
Powershell code to get asset by filter, parse the AssetIds, and delet…

### PR DESCRIPTION
**Pull request by pedrorcs**
Powershell code to get asset by filter, parse the AssetIds, and delete all those assets using the given cleanup time.

### CODE REVIEW

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+.EXAMPLE
+./Get-NB-Assets.ps1 -nbmaster "nb-master.example.com" -username "administrator" -password "secret" -filter "workloadType eq 'VMware'"
+
+#>
+
+param (
+    [string]$nbmaster    = $(throw "Please specify the name of NetBackup master server using -nbmaster parameter."),    
+    [string]$username    = $(throw "Please specify the user name using -username parameter."),
+    [string]$password    = $(throw "Please specify the password using -password parameter."),
+	[string]$filter      = $(throw "Please specify the filter using -filter paramter"),
```
Comment (kk-github-personal) [inline snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Indentation

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 [hunk 2]
```diff
@@ -0,0 +1,168 @@
+<#
+
+.SYNOPSIS
+This sample script demonstrates the use of NetBackup REST API for listing all assets.
+
+.DESCRIPTION
+This script will get a list of asset in the AssetDB that matches the data filter.
+
+.EXAMPLE
+./Get-NB-Assets.ps1 -nbmaster "nb-master.example.com" -username "administrator" -password "secret" -filter "workloadType eq 'VMware'"
```
Comment (kk-github-personal) [inline snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Script is Post-NB-Cleanup-Assets. Example lists something else.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 [hunk 3]
```diff
@@ -0,0 +1,168 @@
+<#
+
+.SYNOPSIS
+This sample script demonstrates the use of NetBackup REST API for listing all assets.
```
Comment (kk-github-personal) [inline snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Script appars to be listing as well as cleaning up the assets.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+.DESCRIPTION
+This script will get a list of asset in the AssetDB that matches the data filter.
```
Comment (kk-github-personal) [inline snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Same as above.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    # This prints all the available attributes
+    #$content_get.data.attributes | Format-Table -AutoSize
+
+    $properties = 
+        @{Label = "Asset ID"; Expression = { $_.id }},
+        @{Label = "ProviderGeneratedId";   Expression = { $_.attributes.providerGeneratedId }},
+        @{Label = "DisplayName";  Expression = { $_.attributes.displayName }},
+        @{Label = "AssetType"; Expression = { $_.attributes.assetType }},
+	    @{Label = "WorkloadType"; Expression = { $_.attributes.workloadType }}
```
Comment (kk-github-personal) [inline snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Indentation.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 2 above)
Comment (kk-github-personal) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Should the usage mention cleanupTime as well?

File: snippets/powershell/README.md [hunk 6]
```diff
@@ -15,4 +15,5 @@ These scripts are only meant to be used as a reference. If you intend to use the
 
 Use the following commands to run the PowerShell samples.
 - `.\Get-NB-Images.ps1 -nbmaster <masterServer> -username <username> -password <password>`
-- `.\Get-NB-Jobs.ps1 -nbmaster <masterServer> -username <username> -password <password>`
+- `.\Get-NB-Jobs.ps1 -nbmaster <masterServer> -username <username> -password <password>
+- `.\Post-NB-Cleanup-Assets.ps1 -nbmaster <masterServer> -username <username> -password <password> -filter <workloadType eq 'Oracle'> -cleanupTime 2018-06-29T15:18:45.678Z
```
Comment (kk-github-personal) [inline snippets/powershell/README.md]: Better to give VMware or Cloud as an example.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 3 above)
Comment (pedrorcs) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Thanks for pointing out. I edited those comments to reflect the script properly.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 4 above)
Comment (pedrorcs) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: I edited those comments as well.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 2 above)
Comment (pedrorcs) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Yes. I fixed.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 1 above)
Comment (pedrorcs) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Fixed.

File: snippets/powershell/Post-NB-Cleanup-Assets.ps1 (hunk 5 above)
Comment (pedrorcs) [reply snippets/powershell/Post-NB-Cleanup-Assets.ps1]: Fixed.

**Comment by pedrorcs** [comment]
Applied the fixes suggested.

File: snippets/powershell/README.md [hunk 7]
```diff
@@ -15,4 +15,5 @@ These scripts are only meant to be used as a reference. If you intend to use the
 
 Use the following commands to run the PowerShell samples.
 - `.\Get-NB-Images.ps1 -nbmaster <masterServer> -username <username> -password <password>`
-- `.\Get-NB-Jobs.ps1 -nbmaster <masterServer> -username <username> -password <password>`
+- `.\Get-NB-Jobs.ps1 -nbmaster <masterServer> -username <username> -password <password>
```
Comment (kk-github-personal) [inline snippets/powershell/README.md]: There is a minor issue with the backticks. Please fix that, rest looks good.

**Review by kk-github-personal — DISMISSED** [review]

**Review by jaya-korlepara — DISMISSED** [review]

File: snippets/powershell/README.md (hunk 7 above)
Comment (pedrorcs) [reply snippets/powershell/README.md]: Fixed. I added to the end of this line the back-tick. Also, I added the the new line of the asset cleanup script I introduced.

**Review by jaya-korlepara — APPROVED** [review]

**Pull request merged**